compiler/AST/ImportStmt.cpp

```diff
@@ -0,0 +1,174 @@
+  }
+}
+
+void ImportStmt::accept(AstVisitor* visitor) {
+  visitor->visitImportStmt(this);
+}
+
+void ImportStmt::verify() {
+  Expr::verify();
+
+  if (astTag != E_ImportStmt) {
+    INT_FATAL(this, "Bad NamedExpr::astTag");
+  }
+
+  if (src == NULL) {
+    INT_FATAL(this, "Bad ImportStmt::src");
+  }
+
+  verifyNotOnList(src);
+}
+
+//
+// Resolve the module symbol referred to by the ImportStmt
+//
+void ImportStmt::scopeResolve(ResolveScope* scope) {
+  /*
+  // 2017-05-28: isValid() does not currently return on failure
+  if (isValid(src) == true) { */
+  // isValid should be re-enabled when we support general exprs in imports
+  // As it is today, we won't encounter the situations it is preventing.
+  // There is an else branch to turn on as well
+    // 2017/05/28 The parser inserts a normalized UseStmt of ChapelBase
+    if (isSymExpr(src)) {
+      INT_FATAL("This should only happen for a UseStmt");
+
+    } else if (Symbol* sym = scope->lookup(src, /*isUse=*/ true)) {
+      SET_LINENO(this);
+
+      if (ModuleSymbol* modSym = toModuleSymbol(sym)) {
+        if (modSym->defPoint->parentSymbol != theProgram) {
+          USR_FATAL_CONT(this, "unable to import nested modules");
+        }
+        scope->enclosingModule()->moduleUseAdd(modSym);
+
+        updateEnclosingBlock(scope, sym);
+
+      } else {
+        if (sym->isImmediate() == true) {
+          USR_FATAL(this,
+                    "'import' statements must refer to module symbols "
+                    "(e.g., 'import <module>;')");
+
+        } else if (sym->name != NULL) {
+          USR_FATAL_CONT(this,
+                         "'import' of non-module symbol %s",
+                         sym->name);
+          USR_FATAL_CONT(sym,  "Definition of symbol %s", sym->name);
+          USR_STOP();
+
+        } else {
+          USR_FATAL(this, "'import' of non-module symbol");
+        }
+      }
+    } else {
+      if (UnresolvedSymExpr* import = toUnresolvedSymExpr(src)) {
+        USR_FATAL(this, "Cannot find module '%s'", import->unresolved);
+      } else {
+        USR_FATAL(this, "Cannot find module");
+      }
+    }
+    /*
+  } else {
+    INT_ASSERT(false);
+    }*/
+}
+
+//
+// Go into the module referred to by the import statement
+//
+BaseAST* ImportStmt::getSearchScope() const {
+  BaseAST* retval = NULL;
+
+  if (SymExpr* se = toSymExpr(src)) {
+    if (ModuleSymbol* module = toModuleSymbol(se->symbol())) {
+      retval = module->block;
+
+    } else {
+      INT_FATAL(this, "Import invalid, not applied to module");
+    }
+
+  } else {
+    INT_FATAL(this, "getSearchScope called before this import was processed");
+  }
+
+  return retval;
+}
+
+//
+// Returns the module symbol if the name provided matches the module imported
+//
+Symbol* ImportStmt::checkIfModuleNameMatches(const char* name) {
+  if (SymExpr* se = toSymExpr(src)) {
+    if (ModuleSymbol* modSym = toModuleSymbol(se->symbol())) {
+      if (strcmp(name, se->symbol()->name) == 0) {
```

Comment:
I would expect it to assume that `name` is an `astr` and then use pointer comparison here